tests/testsuite/build.rs

```diff
@@ -2841,6 +2989,112 @@ fn cargo_platform_specific_dependency() {
     p.cargo("test").run();
 }
 
+#[cargo_test]
+fn cargo_platform_specific_dependency_build_dependencies_conflicting_warning() {
+    let host = rustc_host();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            &format!(
+                r#"
+                    [project]
+                    name = "foo"
+                    version = "0.5.0"
+                    authors = ["wycats@example.com"]
+                    build = "build.rs"
+
+                    [target.{host}.dependencies]
+                    dep = {{ path = "dep" }}
+                    [target.{host}.build-dependencies]
+                    build = {{ path = "build" }}
+                    [target.{host}.build_dependencies]
+                    build = {{ path = "build" }}
+                    [target.{host}.dev-dependencies]
+                    dev = {{ path = "dev" }}
+                "#,
+                host = host
+            ),
+        )
+        .file("src/main.rs", "extern crate dep; fn main() { dep::dep() }")
+        .file(
+            "tests/foo.rs",
+            "extern crate dev; #[test] fn foo() { dev::dev() }",
+        )
+        .file(
+            "build.rs",
+            "extern crate build; fn main() { build::build(); }",
+        )
+        .file("dep/Cargo.toml", &basic_manifest("dep", "0.5.0"))
+        .file("dep/src/lib.rs", "pub fn dep() {}")
+        .file("build/Cargo.toml", &basic_manifest("build", "0.5.0"))
+        .file("build/src/lib.rs", "pub fn build() {}")
+        .file("dev/Cargo.toml", &basic_manifest("dev", "0.5.0"))
+        .file("dev/src/lib.rs", "pub fn dev() {}")
+        .build();
+
+    p.cargo("build")
+        .with_stderr_contains(
+            format!("[WARNING] found both `build-dependencies` and `build_dependencies` are set in the `{}` platform target", host),
+        )
+            .run();
+
+    assert!(p.bin("foo").is_file());
+    p.cargo("test").run();
+}
+
+#[cargo_test]
+fn cargo_platform_specific_dependency_dev_dependencies_conflicting_warning() {
+    let host = rustc_host();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            &format!(
+                r#"
+                    [project]
+                    name = "foo"
+                    version = "0.5.0"
+                    authors = ["wycats@example.com"]
+                    build = "build.rs"
+
+                    [target.{host}.dependencies]
+                    dep = {{ path = "dep" }}
+                    [target.{host}.build-dependencies]
+                    build = {{ path = "build" }}
```

Comment:
Is this intentional having `dependencies` and `build-dependencies` here while testing aginst `dev-dependencies`?

Reply:
I will simplify it.